boards/arm/96b_meerkat96/pinmux.c

```diff
@@ -1,55 +0,0 @@
-/*
- * Copyright (c) 2019, Linaro Ltd.
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#include <zephyr/init.h>
-#include "device_imx.h"
-
-static int meerakt96_pinmux_init(const struct device *dev)
-{
-	ARG_UNUSED(dev);
-
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio1), okay)
-	/* GPIO1_IO04 Mux Config */
-	IOMUXC_LPSR_SW_MUX_CTL_PAD_GPIO1_IO04 = 0;
-	IOMUXC_LPSR_SW_PAD_CTL_PAD_GPIO1_IO04 = 0;
-	/* GPIO1_IO05 Mux Config */
-	IOMUXC_LPSR_SW_MUX_CTL_PAD_GPIO1_IO05 = 0;
-	IOMUXC_LPSR_SW_PAD_CTL_PAD_GPIO1_IO05 = 0;
-	/* GPIO1_IO06 Mux Config */
-	IOMUXC_LPSR_SW_MUX_CTL_PAD_GPIO1_IO06 = 0;
-	IOMUXC_LPSR_SW_PAD_CTL_PAD_GPIO1_IO07 = 0;
-	/* GPIO1_IO07 Mux Config */
-	IOMUXC_LPSR_SW_MUX_CTL_PAD_GPIO1_IO07 = 0;
-	IOMUXC_LPSR_SW_PAD_CTL_PAD_GPIO1_IO07 = 0;
```

Comment:
How should these GPIO's be handled in the pinctrl dts files

Reply:
The GPIO driver should handle these settings. Since these MUX selections are all selecting the GPIO mode for these pins, the device driver or application using these GPIO pins should call `gpio_pin_configure`, which will apply these pinmux settings. So these pinmux settings will not longer be applied at boot, but on an as needed basis by calls to `gpio_pin_configure`